src/openhuman/runtime_pool/pool_worker.py

```
@@ -0,0 +1,174 @@
+    if _HAVE_ALARM and timeout_ms and timeout_ms > 0:
+        def _on_alarm(_signum, _frame):
+            raise _JobTimeout()
+
+        signal.signal(signal.SIGALRM, _on_alarm)
+        signal.setitimer(signal.ITIMER_REAL, timeout_ms / 1000.0)
+        armed = True
+
+    try:
+        # Fresh globals per job so top-level names don't leak between runs.
+        g = {"__name__": "__main__", "__builtins__": __builtins__}
+        exec(compile(code, "<inline>", "exec"), g, g)
```

Ruff (0.15.21)

[error] Use of `exec` detected

(S102)